render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java

```diff
@@ -137,6 +148,61 @@ private void renderJPEG(final ImageSegment imageSegment, final Graphics2D target
         });
     }
 
+    private void renderJPEG2k(final ImageSegment imageSegment, final Graphics2D targetGraphic) throws IOException {
+        final ImageReader reader = getImageReader("image/jp2");
+        reader.setInput(imageSegment.getData(), true, true);
+        final ImageReadParam param = reader.getDefaultReadParam();
+
+        if (ImageRepresentation.MULTIBAND.equals(imageSegment.getImageRepresentation())) {
+            final int[] sourceBands = getSourceBands(imageSegment);
+            param.setSourceBands(sourceBands);
+        }
+
+        processBlocks(imageSegment, (r, c) -> {
+                    Rectangle rect = new Rectangle((int) (c * imageSegment.getNumberOfPixelsPerBlockVertical()),
+                            (int) (r * imageSegment.getNumberOfPixelsPerBlockHorizontal()),
```

Comment:
I'm having trouble convincing myself that this logic is right for non-square blocks. Think of an image that is 400 pixels wide by 300 pixels high. Assume each block is 200 pixels wide (horizontal) by 100 pixels high (vertical). So its 2 blocks wide and 3 blocks high. For block (0,0) it doesn't matter. But it looks like (0,1) will start at (100, 0), but I think that should be (200,).  Maybe see if you can find a test case with non-square blocks?